syntax = "proto3";
message missile_details{
    string missile_type = 1;
    int32 x = 2;
    int32 y = 3;
    int32 t = 4;
}
message survival_response{
    bool is_alive = 1;
}

message Empty{}

service DefenseNotification{
    rpc launch_missile(missile_details) returns (Empty);
}
service CommanderNotification{
    rpc missile_notification(missile_details) returns (Empty);
}
service SoldierNotification{
    rpc notify_soldier(missile_details) returns (Empty);
}
service SoldierSurvival{
    rpc notify_alive_soldier(missile_details) returns (Empty);
    rpc soldier_status(Empty) returns (survival_response);
}

